src/scenes/Jarjestajat/Jarjestaja/Hakemukset/Muutospyynto/components/MuutospyyntoWizard.js

```diff
@@ -222,33 +237,36 @@ const MuutospyyntoWizard = props => {
 
   const getFiles = () => {
     // Gets all attachment data from changeObjects
-    const allAttachments = (
-      R.path(["yhteenveto", "yleisettiedot"], changeObjects) || []
-    ).concat(
-      (R.path(["yhteenveto", "yleisettiedot"], changeObjects) || []).concat(
-        (
-          R.path(["yhteenveto", "hakemuksenliitteet"], changeObjects) || []
-        ).concat(R.path(["taloudelliset", "liitteet"], changeObjects) || [])
+    const allAttachments = combineArrays([
+      R.path(["yhteenveto", "yleisettiedot"], changeObjects) || [],
+      R.path(["yhteenveto", "yleisettiedot"], changeObjects) || [],
+      R.path(["yhteenveto", "hakemuksenliitteet"], changeObjects) || [],
+      R.path(["taloudelliset", "liitteet"], changeObjects) || [],
+      R.path(["perustelut", "liitteet"], changeObjects) || [],
+      R.flatten(
+        R.path(
+          ["perustelut", "koulutukset", "kuljettajakoulutukset"],
+          changeObjects
+        ) || []
       )
-    );
-    // Returns only binary files
-    let attachments;
+    ]);
+    // Returns only attachments
+    let attachments = [];
     if (allAttachments) {
-      attachments = R.map(obj => {
-        if (obj.properties.attachments)
-          return R.map(file => {
-            return file;
+      R.forEachObjIndexed(obj => {
+        if (obj.properties.attachments) {
+          R.forEachObjIndexed(file => {
+            attachments.push(file);
           }, obj.properties.attachments);
-        else return null;
+        }
       }, allAttachments);
```

Comment:
R.filter can be used here for cleaner implementation.

Reply:
->Refactoring later as ths also works.